### ISSUE TITLE
Display of fields

### ISSUE THREAD

**Issue by gavanderhoorn** [UniversalRobots/Universal_Robots_ROS_as_a_Service_URCap#24]
I noticed this while looking at the *Actions* documentation (as just announced by @fmauch), but it's the same for regular messages, so here's a screenshot of that:

![](https://raw.githubusercontent.com/UniversalRobots/Universal_Robots_ROS_as_a_Service_URCap/master/doc/resources/tutorial/13.png)

the current implementation draws a fine line between all the fields of the message (under the *Data* section in the screenshot).

To me, this implies there is some sort of relation between these fields (as if the line 'ties them together').

In a sense, there is of course such a relationship: they are all fields in the same message, so *children*. The fact the line doesn't go to a shared 'root' confuses me as a user.

Would it perhaps be an idea to make the *Data* heading left-aligned in the panel, and then make the line appear to start right under that heading? Something like this (ignore the ms paint skills):

![image](https://user-images.githubusercontent.com/4550046/116682345-6e9b3080-a9ae-11eb-93bf-a1942b9ca7f4.png)

If this is already done, and I've missed it, please `/ignore`.

**Comment by gavanderhoorn** [comment]
It might even be nice to change *Data* to *Message fields* or something similar?

**Comment by fmauch** [comment]
Yes, we have an issue about that in our internal big tracker, as well, though you raised additional aspects of this. Let me come back to that once I migrate all internal issues to GitHub.

**Comment by urfeex** [comment]
ROS 1 is EOL
Publish and subscribing to ROS 2 messages is natively [supported in PolyScope X](https://docs.universal-robots.com/polyscopex-ros2/v10.12/index.html)
